sdk/ml/azure-ai-ml/azure/ai/ml/operations/_component_operations.py

```diff
@@ -767,7 +767,9 @@ def _resolve_arm_id_or_upload_dependencies(self, component: Component) -> None:
             self._all_operations, self._operation_scope, self._operation_config
         ).get_asset_arm_id
 
-        self._resolve_dependencies_for_component(component, resolver)
+        self._resolve_dependencies_for_component(
```

Comment:
why this solves issue for @dyanmic?

Reply:
Actually it's not a dynamic specific issue but existed for all pipeline component. 1 possible reason is that our customer rarely uses the combination I mentioned - static input inside a subgraph.